host/onebox.go

```diff
@@ -143,31 +151,33 @@ type (
 // NewCadence returns an instance that hosts full cadence in one process
 func NewCadence(params *CadenceParams) Cadence {
 	return &cadenceImpl{
-		logger:                  params.Logger,
-		clusterMetadata:         params.ClusterMetadata,
-		persistenceConfig:       params.PersistenceConfig,
-		dispatcherProvider:      params.DispatcherProvider,
-		messagingClient:         params.MessagingClient,
-		metadataMgr:             params.MetadataMgr,
-		metadataMgrV2:           params.MetadataMgrV2,
-		visibilityMgr:           params.VisibilityMgr,
-		shardMgr:                params.ShardMgr,
-		historyMgr:              params.HistoryMgr,
-		historyV2Mgr:            params.HistoryV2Mgr,
-		taskMgr:                 params.TaskMgr,
-		executionMgrFactory:     params.ExecutionMgrFactory,
-		shutdownCh:              make(chan struct{}),
-		clusterNo:               params.ClusterNo,
-		enableWorkerService:     params.EnableWorker,
-		enableEventsV2:          params.EnableEventsV2,
-		enableVisibilityToKafka: params.EnableVisibilityToKafka,
-		blobstoreClient:         params.Blobstore,
-		historyConfig:           params.HistoryConfig,
+		logger:                     params.Logger,
+		clusterMetadata:            params.ClusterMetadata,
+		persistenceConfig:          params.PersistenceConfig,
+		dispatcherProvider:         params.DispatcherProvider,
+		messagingClient:            params.MessagingClient,
+		metadataMgr:                params.MetadataMgr,
+		metadataMgrV2:              params.MetadataMgrV2,
+		visibilityMgr:              params.VisibilityMgr,
+		shardMgr:                   params.ShardMgr,
+		historyMgr:                 params.HistoryMgr,
+		historyV2Mgr:               params.HistoryV2Mgr,
+		taskMgr:                    params.TaskMgr,
+		executionMgrFactory:        params.ExecutionMgrFactory,
+		shutdownCh:                 make(chan struct{}),
+		clusterNo:                  params.ClusterNo,
+		enableEventsV2:             params.EnableEventsV2,
+		esConfig:                   params.ESConfig,
+		esClient:                   params.ESClient,
+		enableReadVisibilityFromES: params.EnableReadVisibilityFromES,
+		blobstoreClient:            params.Blobstore,
+		historyConfig:              params.HistoryConfig,
+		workerConfig:               params.WorkerConfig,
 	}
 }
 
 func (c *cadenceImpl) enableWorker() bool {
```

Comment:
Is this still needed?

Reply:
It's still used several places in the code.